mmdet/core/evaluation/mean_ap.py

```diff
@@ -162,15 +171,146 @@ def tpfp_imagenet(det_bboxes,
+                            fp[k, i] = 1
+            if ioas.shape[1] <= 0:
+                return tp, fp, det_bboxes
+
+        if ioas.shape[1] > 0:
+            ignore_det_bboxes = np.zeros(num_dets, dtype=bool)
+            match_group_of = np.zeros((num_scales, num_dets), dtype=bool)
+            tp_group = np.zeros((num_scales, num_gts_group), dtype=np.float32)
+            fp_group = np.zeros((num_scales, num_gts_group), dtype=np.float32)
+            ioas_max = ioas.max(axis=1)
+            # for each det, which gt overlaps most with it
+            ioas_argmax = ioas.argmax(axis=1)
+            # sort all dets in descending order by scores
+            sort_inds = np.argsort(-det_bboxes[:, -1])
+            for k, (min_area, max_area) in enumerate(area_ranges):
+                box_is_covered = tp[k]
+                # group_covered = np.zeros(num_gts_group, dtype=bool)
+                # if no area range is specified, gt_area_ignore is all False
+                if min_area is None:
+                    gt_area_ignore = np.zeros_like(gt_ignore_inds, dtype=bool)
+                else:
+                    gt_areas = (gt_bboxes[:, 2] - gt_bboxes[:, 0]) * (
+                        gt_bboxes[:, 3] - gt_bboxes[:, 1])
+                    gt_area_ignore = (gt_areas < min_area) | (
+                        gt_areas >= max_area)
+                for i in sort_inds:
+                    matched_gt = ioas_argmax[i]
+                    if not box_is_covered[i]:
+                        if ioas_max[i] >= ioa_thr:
+                            if not (gt_ignore_inds[matched_gt]
+                                    or gt_area_ignore[matched_gt]):
+                                if not tp_group[k, matched_gt]:
+                                    tp_group[k, matched_gt] = 1
+                                    match_group_of[k, i] = True
+                                else:
+                                    ignore_det_bboxes[i] = True
+                                    match_group_of[k, i] = True
+                            else:
+                                fp_group[k, matched_gt] = 1
+            tps = []
+            fps = []
+            for i in range(num_scales):
+                tps.append(
+                    np.concatenate((tp[i][~match_group_of[i]], tp_group[i])))
+                fps.append(
+                    np.concatenate((fp[i][~match_group_of[i]], fp_group[i])))
+
+            tp = np.vstack(tps)
+            fp = np.vstack(fps)
+            det_bboxes = det_bboxes[~ignore_det_bboxes]
+
+        return tp, fp, det_bboxes
 
 
 def tpfp_default(det_bboxes,
                  gt_bboxes,
                  gt_bboxes_ignore=None,
+                 gt_bboxes_group_of=None,
                  iou_thr=0.5,
+                 ioa_thr=None,
                  area_ranges=None,
-                 use_legacy_coordinate=False):
+                 use_legacy_coordinate=False,
+                 use_group_of=False):
```

Comment:
docstring.
And what is the meaning of 'of'

Reply:
`group of` is a name which used in OpenImages annotations